crates/moon/src/cli/new.rs

```diff
@@ -88,105 +63,43 @@ pub fn run_new(_cli: &UniversalFlags, cmd: NewSubcommand) -> anyhow::Result<i32>
-
-    let (target_dir, user, name, license) =
-        if let (Some(path), Some(user), Some(name)) = (path, user, name) {
-            (path, user, name, license.map(|s| s.to_string()))
-        } else {
-            let tmp = dialoguer::Input::<String>::new()
-                .with_prompt("Enter the path to create the project (. for current directory)")
-                .default("my-project".to_string())
-                .show_default(true)
-                .validate_with(|input: &String| -> Result<(), String> {
-                    let p = input.trim();
-                    let dot = p == ".";
-
-                    let p = PathBuf::from(p);
-
-                    if p.exists() {
-                        if p.join(MOON_MOD_JSON).exists() {
-                            Err(format!(
-                                "A MoonBit project already exists in `{}`.",
-                                p.display()
-                            ))
-                        } else {
-                            if !dot {
-                                eprintln!(
-                                    "{}: The directory is already exists.",
-                                    "Warning".yellow().bold(),
-                                );
-                            };
-                            Ok(())
-                        }
-                    } else {
-                        Ok(())
-                    }
-                })
-                .interact()?;
-            let path = PathBuf::from(tmp);
-
-            let existing_username = get_existing_username();
-            let default_username = existing_username.unwrap_or_else(|| "username".to_string());
-
-            let username = dialoguer::Input::<String>::new()
-                .with_prompt("Enter your username")
-                .default(default_username)
-                .validate_with(|input: &String| -> Result<(), String> { validate_username(input) })
-                .show_default(true)
-                .interact()?;
-
-            let package_name = dialoguer::Input::<String>::new()
-                .with_prompt("Enter your project name")
-                .default("hello".to_string())
-                .show_default(true)
-                .interact()?;
-
-            let license = dialoguer::Input::<String>::new()
-                .with_prompt("Enter your license")
-                .default("Apache-2.0".to_string())
-                .show_default(true)
-                .interact()?;
-
-            (path, username, package_name, Some(license))
-        };
-
-    if target_dir.exists() && target_dir.join(MOON_MOD_JSON).exists() {
-        bail!(
-            "A MoonBit project already exists in `{}`.",
-            target_dir.display()
-        );
+    let existing_username = get_existing_username();
+    let username = cmd
+        .user
+        .or(existing_username)
+        .unwrap_or_else(|| "username".to_string());
+    validate_username(&username).map_err(|e| anyhow::anyhow!(e))?;
```

Comment:
You can just `validate()?`, `?` runs `.into()` for all errors, which is the same operation as what you are writing.